### PR TITLE
Removed Default trait from storage.

### PR DESCRIPTION
**Pull request by kubaplas**

### CODE REVIEW

File: dao-utils/src/parts/variable.rs [hunk 1]
```diff
@@ -25,9 +26,14 @@ impl<T: Default + FromBytes + ToBytes + CLTyped> Variable<T> {
...
-    pub fn get(&self) -> T {
-        get_key(&self.name).unwrap_or_default()
+    /// Read from the storage or return none
+    pub fn get(&self) -> Option<T> {
+        get_key(&self.name)
+    }
+
+    /// Read from the storage or revert
+    pub fn get_or_revert(&self) -> T {
+        get_key(&self.name).unwrap_or_revert()
```
Comment (kpob) [inline dao-utils/src/parts/variable.rs:36]: Do we want to just revert without any specific error? I encounterd this many times in this pr. Honestly, it's fine to me, but what if we revert with some generic error in such cases?

File: dao-contracts/src/voting/kyc_info.rs [hunk 2]
```diff
@@ -21,6 +21,7 @@ impl KycInfo {
         self.kyc_token
             .get()
             .unwrap_or_revert_with(Error::VariableValueNotSet)
+            .unwrap_or_revert_with(Error::VariableValueNotSet)
```
Comment (kpob) [inline dao-contracts/src/voting/kyc_info.rs:24]: Shouldn't be unwraped twice, make the `kyc_token` variable non-optional

File: dao-contracts/src/voting/onboarding_info.rs [hunk 3]
```diff
@@ -21,6 +21,7 @@ impl OnboardingInfo {
         self.va_token
             .get()
             .unwrap_or_revert_with(Error::VariableValueNotSet)
+            .unwrap_or_revert_with(Error::VariableValueNotSet)
```
Comment (kpob) [inline dao-contracts/src/voting/onboarding_info.rs:24]: Shouldn't be unwraped twice, make the kyc_token variable non-optional

File: dao-erc20/src/erc20.rs [hunk 4]
```diff
@@ -48,23 +48,23 @@ impl ERC20Interface for ERC20 {
...
     }
 
     fn decimals(&self) -> u8 {
-        self.decimals.get()
+        self.decimals.get_or_revert()
     }
 
     fn total_supply(&self) -> U256 {
-        self.total_supply.get()
+        self.total_supply.get_or_revert()
```
Comment (kpob) [inline dao-erc20/src/erc20.rs:63]: Why to revert? I would return the default value.

File: dao-utils/src/parts/collection.rs [hunk 5]
```diff
@@ -46,17 +47,16 @@ impl<T: ToBytes + FromBytes + CLTyped + Default + PartialEq + Debug + Hash> Orde
...
-        self.length.get()
+        self.length.get().unwrap_or(0)
     }
 
     fn move_item(&mut self, from: u32, to: u32) {
-        let value = self.values.get(from).unwrap();
-        self.values.set(to, value);
-        self.values.unset(from);
+        let value = self.values.get(from);
+        self.values.set(to, value.unwrap_or_revert());
```
Comment (kpob) [inline dao-utils/src/parts/collection.rs:55]: Why did you remove `self.values.unset(from)`? If I'm right, without it, we've got two occurences of the `from` value.

File: dao-erc20/src/erc20.rs (hunk 4 above)
Comment (kubaplas) [reply dao-erc20/src/erc20.rs:63]: It shouldn't ever happen, as it is set during the init, so it really doesn't matter

File: dao-utils/src/parts/collection.rs (hunk 5 above)
Comment (kubaplas) [reply dao-utils/src/parts/collection.rs:55]: Don't be so interested or you'll get catface :cat:

**Review by kpob — APPROVED** [review]
